BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsClient.kt

```diff
@@ -16,14 +15,16 @@ import java.util.concurrent.TimeUnit
 
 @Suppress("SwallowedException", "TooGenericExceptionCaught")
 internal class AnalyticsClient @VisibleForTesting constructor(
```

Comment:
Outside of this PR's scope: What are your thoughts on using default values in the constructor instead of defining multiple constructors? 

This entire class is internal, so we wouldn't be leaking anything to the merchant. We could also remove the `@VisibleForTesting` annotation.

This might be a solution to the DI issue I ran into in my other PR.

Reply:
We could yeah. The only reason I've done this in the past is to discourage use of the `@VisibleForTesting` constructor. But we could also enforce this by convention. I agree it's cleaner with default parameter values.